stdpopsim/models.py

```diff
@@ -5,6 +5,11 @@
 import textwrap
 
 import msprime
+import warnings
+
+# DeprecationWarning is filtered by default. The following ensures that these
+# warnings are visible to the user.
+warnings.simplefilter("always", DeprecationWarning)
```

Comment:
Is this a bad idea? Without changing warning filtering settings, the deprecation warning for `model.get_samples` will be filtered by default and not visible to the user. On the other hand, DeprecationWarnings from other packages that are intended for developers will show up. Maybe it would be better to define a custom deprecation warning in `stdpopsim/warning_categories.py`?

Do you have a suggestion for this @jeromekelleher?

Reply:
I'm not sure - @grahamgower has a better handle on how we're using warnings here I think?

Reply:
Hmm, sorry, I actually don't know how to do sane deprecation warnings in Python. The only thing I do know is that the facilities provided in Python are confusing and that folks complain that warnings are either too noisy and thus ignored, or alternately never reach the right people at the right time.

Reply:
ok thanks for the input -- I'll make a `DeprecatedFeatureWarning` that inherits from `UserWarning` into stdpopsim/warning_categories.py ... will be visible to user and won't alter warning behavior from other modules.